nomt/src/store.rs

```diff
@@ -0,0 +1,116 @@
+//! A wrapper around RocksDB for avoiding prolifiration of RocksDB-specific code.
+
+use nomt_core::{
+    page_id::PageId,
+    trie::{KeyPath, Node, TERMINATOR},
+};
+use rocksdb::{ColumnFamilyDescriptor, DBWithThreadMode, MultiThreaded, WriteBatch};
+use std::sync::Arc;
+
+static FLAT_KV_CF: &str = "flat_kv";
+static PAGES_CF: &str = "pages";
+static METADATA_CF: &str = "metadata";
+
+/// This is a lightweight handle and can be cloned cheaply.
+#[derive(Clone)]
+pub struct Store {
+    shared: Arc<Shared>,
+}
+
+struct Shared {
+    // TODO: investigate if we actually need the `Multithreaded`. According to docs and source code,
+    //       it seems that it's only needed in the case of changing the cf set during the lifetime.
+    //
+    //       We don't do that, so we might be able to use `SingleThreaded` instead.
+    db: DBWithThreadMode<MultiThreaded>,
```

Comment:
FWIW polkadot-sdk is quite a lot of production evidence that accessing the DB from multiple threads is A-OK.

ref
https://github.com/paritytech/parity-common/blob/master/kvdb-rocksdb/src/lib.rs#L248 
https://docs.rs/rocksdb/latest/rocksdb/type.DB.html

Reply:
https://github.com/thrumdev/nomt/pull/43